plugins/inputs/bigbluebutton/bigbluebutton.go

```diff
@@ -0,0 +1,239 @@
+	return hash.Sum(nil)
+}
+
+func (b *BigBlueButton) getURL(apiCallName string) string {
+	endpoint := fmt.Sprintf("%s/api/%s", b.PathPrefix, apiCallName)
+	return fmt.Sprintf("%s%s?checksum=%x", b.URL, endpoint, b.checksum(apiCallName))
+}
+
+// Call BBB server api
+func (b *BigBlueButton) api(url string) ([]byte, error) {
+	request, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if b.Username != "" || b.Password != "" {
+		request.SetBasicAuth(b.Username, b.Password)
+	}
+
+	resp, err := b.client.Do(request)
+
+	if err != nil || resp.StatusCode != 200 {
+		return nil, fmt.Errorf("error getting bbb metrics: %s status %d", err, resp.StatusCode)
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func (b *BigBlueButton) gatherMeetings(acc telegraf.Accumulator) error {
+	body, err := b.api(b.getMeetingsURL)
+	if err != nil {
+		return err
+	}
+
+	var response MeetingsResponse
+	err = xml.Unmarshal(body, &response)
+	if err != nil {
+		return err
+	}
+
+	record := map[string]uint64{
+		"active_recording":        0,
+		"listener_count":          0,
+		"participant_count":       0,
+		"video_count":             0,
+		"voice_participant_count": 0,
+	}
+
+	if response.MessageKey == "noMeetings" {
+		acc.AddFields("bigbluebutton_meetings", toStringMapInterface(record), make(map[string]string))
+		return nil
+	}
+
+	for i := 0; i < len(response.Meetings.Values); i++ {
+		meeting := response.Meetings.Values[i]
+		record["participant_count"] += meeting.ParticipantCount
+		record["listener_count"] += meeting.ListenerCount
+		record["voice_participant_count"] += meeting.VoiceParticipantCount
+		record["video_count"] += meeting.VideoCount
+		if meeting.Recording == true {
+			record["active_recording"]++
+		}
+	}
+
+	acc.AddFields("bigbluebutton_meetings", toStringMapInterface(record), make(map[string]string))
+	return nil
+}
+
+func (b *BigBlueButton) gatherRecordings(acc telegraf.Accumulator) error {
+	body, err := b.api(b.getRecordingsURL)
+	if err != nil {
+		return err
+	}
+
+	var response RecordingsResponse
+	err = xml.Unmarshal(body, &response)
+	if err != nil {
+		return err
+	}
+
+	record := map[string]uint64{
+		"recordings_count":           0,
+		"published_recordings_count": 0,
+	}
+
+	if response.MessageKey == "noRecordings" {
+		acc.AddFields("bigbluebutton_recordings", toStringMapInterface(record), make(map[string]string))
+		return nil
+	}
+
+	for i := 0; i < len(response.Recordings.Values); i++ {
+		recording := response.Recordings.Values[i]
+		record["recordings_count"]++
+		if recording.Published {
+			record["published_recordings_count"]++
+		}
+	}
+
+	acc.AddFields("bigbluebutton_recordings", toStringMapInterface(record), make(map[string]string))
+	return nil
+}
+
+func toStringMapInterface(in map[string]uint64) map[string]interface{} {
+	var m = map[string]interface{}{}
```

Comment:
Directly allocate with the correct capacity to save some reallocations.
```suggestion
	m := make(map[string]interface{}, len(in))
```

Reply:
Done